src/components/radio-group/radio-group.component.ts

```diff
@@ -191,17 +191,23 @@ export default class SlRadioGroup extends ShoelaceElement implements ShoelaceFor
     event.preventDefault();
   }
 
-  private handleLabelClick() {
+  private handleFocus(options?: FocusOptions) {
     const radios = this.getAllRadios();
     const checked = radios.find(radio => radio.checked);
-    const radioToFocus = checked || radios[0];
+    const firstEnabledRadio = radios.find(radio => !radio.disabled);
+    const radioToFocus = checked || firstEnabledRadio;
 
-    // Move focus to the checked radio (or the first one if none are checked) when clicking the label
+    // Call focus for the checked radio
+    // If no radio is checked, focus the first one that is not disabled
     if (radioToFocus) {
-      radioToFocus.focus();
+      radioToFocus.focus(options);
     }
   }
```

Comment:
This can probably be moved into the public `focus()` method.

Reply:
The reason I moved this to its own method was that `handleLabelClick` uses the same logic. However you are right, I could call `this.focus()` instead of `handleFocus.focus()` in `handleLabelClick`, removing one method. Will update this in a moment

Reply:
Done